src/dbnode/server/server.go

```diff
@@ -1165,6 +1167,81 @@ func kvWatchEncodersPerBlockLimit(
 	}()
 }
 
+func kvWatchQueryLimit(
+	store kv.Store,
+	logger *zap.Logger,
+	limits limits.QueryLimits,
+) {
+	value, err := store.Get(kvconfig.QueryLimits)
+	if err == nil {
+		protoValue := &kvpb.QueryLimits{}
+		err = value.Unmarshal(protoValue)
+		if err == nil && protoValue != nil {
+			updateQueryLimits(logger, limits, protoValue)
+		}
+	} else if !errors.Is(err, kv.ErrNotFound) {
+		logger.Warn("error resolving query limit", zap.Error(err))
```

Comment:
Hm does watching not work on a key that doesn't exist? Ideal behavior I think would be to allow for non-set key but pick it up if it does get set. Otherwise, maybe we should call `SetIfNotExist` or something to seed the db w/ the initial config

Reply:
Confirmed this works w/o the key. E.g. I get this on startup w/o a key present
```
{"level":"info","ts":1611167411.914984,"msg":"creating etcd watch at revision","watch_key":"_kv/default_env/m3db.query.limits","revision":0}
```
and I think that's what we want because we want by-default this to not be etcd driven.

Reply:
Ah I see, and then the watch lets you can re-add the key to start watching it? If so all good :+1: